libs/block-mesh-manager/src/routes/admin/reports_queue/get_stats.rs

```diff
@@ -0,0 +1,43 @@
+use crate::database::user::get_user_by_id::get_user_opt_by_id;
+use crate::domain::user::UserRole;
+use crate::errors::error::Error;
+use crate::middlewares::authentication::Backend;
+use crate::startup::application::AppState;
+use askama_axum::IntoResponse;
+use axum::extract::State;
+use axum::{debug_handler, Extension, Json};
+use axum_login::AuthSession;
+use http::StatusCode;
+use std::sync::Arc;
+
+#[debug_handler]
+pub async fn handler(
+    State(state): State<Arc<AppState>>,
+    Extension(auth): Extension<AuthSession<Backend>>,
+) -> Result<impl IntoResponse, Error> {
+    let mut transaction = state.pool.begin().await?;
+    let user = auth.user.ok_or(Error::UserNotFound)?;
+    let user = get_user_opt_by_id(&mut transaction, &user.id)
+        .await?
+        .ok_or(Error::UserNotFound)?;
+    if !matches!(user.role, UserRole::Admin) {
+        return Err(Error::Unauthorized);
+    }
+    transaction.commit().await?;
+    if let Some(mut controller) = state
```

Comment:
Again, the controller owns this state which is weird.
The data is stored in the DB, and anyone that wants it simply access the DB.
The ConnectionManager shouldn't own it.

Reply:
Are you saying that we should simply keep the stas in the database in place of locally owned channels?